watson_developer_cloud/common.py

```diff
@@ -0,0 +1,40 @@
+# coding: utf-8
+
+# Copyright 2019 IBM All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import platform
+from .version import __version__
+
+SDK_ANALYTICS_HEADER = 'X-IBMCloud-SDK-Analytics'
+USER_AGENT_HEADER = 'User-Agent'
+SDK_NAME = 'watson-apis-python-sdk'
+
+def get_system_info():
+    return '{0} {1} {2}'.format(platform.system(), # OS
+                                platform.release(), # OS version
+                                platform.python_version()) # Python version
+
+def get_user_agent():
+    return '{0}-{1} {2}'.format(SDK_NAME, __version__, get_system_info())
```

Comment:
In the area of performance... it would probably be better if we only computed this string once the first time the function was called and then cached the value for each subsequent call. That way we're not computing the string EVERY time a REST API is invoked.

Reply:
Good point 👍 I updated it to be in a global variable.